### PR TITLE
trf: init at 4.09.1

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

Tandem Repeats Finder: a program to analyze DNA sequences

homepage
https://tandem.bu.edu/trf/trf.html

repo
https://github.com/Benson-Genomics-Lab/TRF

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/trf/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchFromGitHub, }:
+
+stdenv.mkDerivation rec {
+  pname = "trf";
+  version = "4.09.1";
+
+  src = fetchFromGitHub {
+    owner = "Benson-Genomics-Lab";
+    repo = pname;
```
Comment (sikmir) [inline pkgs/applications/science/biology/trf/default.nix:9]:
```suggestion
    repo = "trf";
```

File: pkgs/applications/science/biology/trf/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-73LypVqBdlRdDCblf9JNZQmS5Za8xpId4ha5GjTJHDo=";
+  };
+
+  meta = with lib; {
+    description = "Tandem Repeats Finder: a program to analyze DNA sequences";
+    homepage = "https://tandem.bu.edu/trf/trf.html";
+    license = licenses.agpl3;
```
Comment (sikmir) [inline pkgs/applications/science/biology/trf/default.nix:17]:
```suggestion
    license = licenses.agpl3Plus;
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 221969` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trf</li>
  </ul>
</details>

File: pkgs/applications/science/biology/trf/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/applications/science/biology/trf/default.nix:17]: I didn't know about agpl3 or later, thanks!

Comment (sikmir) [reply pkgs/applications/science/biology/trf/default.nix:17]: That's from here https://github.com/Benson-Genomics-Lab/TRF/blob/355c1f92b24f6b2e71650eda3549d46f0316e571/src/trf.c#L7-L10: `version 3 of
the License, or (at your option) any later version`.

File: pkgs/applications/science/biology/trf/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/applications/science/biology/trf/default.nix:9]: There are fewer examples of 'repo = pname;' in the nixpkgs code, so it seems better to define it explicitly.
Is there any documentation recommending this?

File: pkgs/applications/science/biology/trf/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchFromGitHub, }:
```
Comment (sikmir) [inline pkgs/applications/science/biology/trf/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub }:
```

File: pkgs/applications/science/biology/trf/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/applications/science/biology/trf/default.nix]: >  since repo and pname are not the same semantically

Exactly, I'll remember it.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 221969` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trf</li>
  </ul>
</details>

File: pkgs/applications/science/biology/trf/default.nix (hunk 3 above)
Comment (natsukium) [reply pkgs/applications/science/biology/trf/default.nix]: done

**Comment by natsukium** [comment]
Thanks for reviewing!

**Pull request merged**